stout/flags.cc

```diff
@@ -201,8 +201,8 @@ void Parser::Parse(
         if (text != parsed_[field].text) {
           errors.insert(
               "Encountered duplicate boolean flag '" + non_negated_name + "' "
-              + (parsed_[field].name != non_negated_name
-                     ? "with flag aliased as '" + parsed_[field].name + "' "
+              + (parsed_[field].name != /*non_negated_name*/ name
+                     ? "with flag aliased as '" + /*parsed_[field].name*/ name + "' "
```

Comment:
Ah, this is for a slightly different case, when there is actually an alias for the flag that is different than the actual name. This can happen because you can specify more than one [name(https://github.com/3rdparty/stout-flags/blob/main/stout/v1/flag.proto#L8) for the flag. You can totally write a test for this case too!

It looks like what you've actually uncovered in your `DuplicateConflict` test is a _different_ error condition than this piece of code where we want to have a better error message. So let's revert this change here and how about adding another test for this case? Then you can add some code to address the `DuplicateConflict` test case you already have (or at least the part of that test case that deals with the `--no-b --b` part of the flags). 
 
```suggestion
              + (parsed_[field].name != non_negated_name
                     ? "with flag aliased as '" + parsed_[field].name + "' "
```

Reply:
Looking forward to the followup PR for this!